external-crates/move/crates/move-compiler/src/linters/combinable_bool_conditions.rs

```diff
@@ -0,0 +1,157 @@
+//! The `CombinableBool` detects and warns about boolean conditions in Move code that can be simplified.
+//! It identifies comparisons that are logically equivalent and suggests more concise alternatives.
+//! This rule focuses on simplifying expressions involving `==`, `<`, `>`, and `!=` operators to improve code readability.
+use move_ir_types::location::Loc;
+
+use crate::{
+    diag,
+    diagnostics::{
+        codes::{custom, DiagnosticInfo, Severity},
+        WarningFilters,
+    },
+    parser::ast::BinOp_,
+    shared::{program_info::TypingProgramInfo, CompilationEnv},
+    typing::{
+        ast::{self as T, UnannotatedExp_},
+        visitor::{TypingVisitorConstructor, TypingVisitorContext},
+    },
+};
+
+use super::{LinterDiagCategory, LINTER_COMB_BOOL, LINT_WARNING_PREFIX};
+
+const COMBINABLE_BOOL_COND_DIAG: DiagnosticInfo = custom(
+    LINT_WARNING_PREFIX,
+    Severity::Warning,
+    LinterDiagCategory::Readability as u8,
+    LINTER_COMB_BOOL,
+    "",
+);
+
+pub struct CombinableBool;
+
+pub struct Context<'a> {
+    env: &'a mut CompilationEnv,
+}
+
+impl TypingVisitorConstructor for CombinableBool {
```

Comment:
We should move this to the CFGIR to take advantage of constant folding and inlining

Reply:
We will need to setup a non-absint visitor for you

Reply:
Is that mean you setup visitor for us or we do it by ourself?